webapp/src/components/systemConsole/expiry/expiry.tsx

```diff
@@ -4,25 +4,34 @@
 import type {ChangeEvent} from 'react';
 import React, {useCallback, useEffect, useState} from 'react';
 
-import type {CustomComponentProps, ExpiryConfig} from 'types/mattermost-webapp';
+import type {CustomSettingChildComponentProp} from 'components/systemConsole/index';
+
+import type {ExpiryConfig} from 'types/mattermost-webapp';
 
 import './style.scss';
 
-const defaultExpiry = '30';
+const Expiry = ({id, setSaveNeeded, onChange, config, setInitialSetting}: CustomSettingChildComponentProp) => {
+    // default values
+    const defaultExpiry = '30';
 
-const Expiry = ({id, setSaveNeeded, onChange, config}: CustomComponentProps) => {
     const [expiryDays, setExpiryDays] = useState<string>(defaultExpiry);
     const [error, setError] = useState<string>('');
 
+    // Set initial value from saved config
     useEffect(() => {
-        // Set initial value from saved config
+        const expiryConfig = config.PluginSettings?.Plugins?.['com.mattermost.user-survey']?.systemconsolesetting?.SurveyExpiry;
 
-        const expiryConfig = config.PluginSettings?.Plugins?.['com.mattermost.user-survey']?.surveyexpiry;
+        const initialSetting: ExpiryConfig = {
+            days: Number.parseInt(defaultExpiry, 10),
+        };
 
         if (expiryConfig?.days) {
             setExpiryDays(expiryConfig.days.toString());
+            initialSetting.days = expiryConfig.days;
         }
```

Comment:
In the previous component, we set the local value always, but here we only set the local state if the config is truthy (so we wouldn't update it if the config is 0).

Is this intended?

Reply:
Not sure where you're referring, but I'm initializing the state with default value already so need of doing it here.

I think you're referring to the `SurveyDateTime` component, there I'm setting state even if false value in config as the default value os not computable in a single line, or rather it cleaner in multiple lines.